File: Leetcode/PY/Leetcode383-赎金信.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mp = defaultdict(int)
        for ch in magazine:
            mp[ch] += 1
        for ch in ransomNote:
            if ch in mp and mp[ch] != 0:
                mp[ch] -= 1
            else:
                return False
        return True

File: Leetcode/Rust/Leetcode383-赎金信.rs
impl Solution {
    pub fn can_construct(ransom_note: String, magazine: String) -> bool {
        let mut mp = vec![0;26];
        for ch in magazine.bytes(){
            let ch = (ch - b'a') as usize;
            mp[ch] += 1;
        }
        for ch in ransom_note.bytes(){
            let ch = (ch - b'a') as usize;      
            mp[ch] -= 1;
            if mp[ch] < 0 {
                return false;
            }
        }
        true
    }
}